cmake/Modules/FindCUDNN.cmake

```diff
@@ -19,39 +19,55 @@
 #  ``CUDNN_LIBRARIES``
 #    The CUDNN libraries.
 #
+#  ``CuDNN::CuDNN``
+#    The CUDNN target
+#
 
 include(FindPackageHandleStandardArgs)
 
-if(NOT CUDNN_INCLUDE_DIR)
-  find_path(CUDNN_INCLUDE_DIR cudnn.h
-    HINTS ${CUDA_HOME} ${CUDA_TOOLKIT_ROOT_DIR} $ENV{cudnn} $ENV{CUDNN}
-    PATH_SUFFIXES cuda/include include)
-endif()
-
-if(NOT CUDNN_LIBRARY)
-  find_library(CUDNN_LIBRARY cudnn
-    HINTS ${CUDA_HOME} ${CUDA_TOOLKIT_ROOT_DIR} $ENV{cudnn} $ENV{CUDNN}
-    PATH_SUFFIXES lib lib64 cuda/lib cuda/lib64 lib/x64)
-endif()
-
+find_path(CUDNN_INCLUDE_DIR NAMES cudnn.h cudnn_v8.h cudnn_v7.h
+  HINTS $ENV{CUDA_PATH} $ENV{CUDA_TOOLKIT_ROOT_DIR} $ENV{cudnn} $ENV{CUDNN} $ENV{CUDNN_ROOT_DIR} /usr/include
```

Comment:
Are $ENV{cudnn} $ENV{CUDNN} official. It looks like a hack.
For Linux $ENV{CUDA_HOME} should be considered.